File: src/java/_1431_Kids_With_the_Greatest_Number_of_Candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int greatest = 0;

        for (int candy : candies)
            greatest = Math.max(greatest, candy);

        List<Boolean> ans = new ArrayList<>();

        for (int candy : candies)
            ans.add(candy + extraCandies >= greatest);

        return ans;
    }
}

File: src/_1431_Kids_With_the_Greatest_Number_of_Candies.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
bool *kidsWithCandies(int *candies, int candiesSize, int extraCandies, int *returnSize)
{
    *returnSize = candiesSize;
    int greatest = 0;

    for (int i = 0; i < candiesSize; i++)
        greatest = greatest > candies[i] ? greatest : candies[i];

    bool *ans = (bool *)malloc(candiesSize * sizeof(bool));

    for (int i = 0; i < candiesSize; i++)
        ans[i] = (candies[i] + extraCandies >= greatest);

    return ans;
}
